packages/cdk/lib/wes_adapter/amazon_genomics/wes/adapters/CromwellWESAdapter.py

```diff
@@ -0,0 +1,686 @@
+      * optionally, if an `options.json` file is found in the root level of the zip, it will be set as `workflowOptions`
+
+    If the object is a `workflow.zip` file containing multiple files, the `workflow.zip` file is set as `workflowDependencies`
+    """
+    s3 = boto3.resource("s3")
+
+    u = urlparse(s3_uri)
+    bucket = s3.Bucket(u.netloc)
+    key = u.path[1:]
+
+    data = dict()
+    files = dict()
+
+    if not key:
+        raise RuntimeError("invalid or missing S3 object key")
+
+    try:
+        file = path.join(localpath, path.basename(key))
+        bucket.download_file(key, file)
+    except botocore.exceptions.ClientError as e:
+        raise RuntimeError(f"invalid S3 object: {e}")
+
+    if path.basename(file) == "workflow.zip":
+        try:
+            props = parse_workflow_zip_file(file, workflow_type)
+        except Exception as e:
+            raise RuntimeError(f"{s3_uri} is not a valid workflow.zip file: {e}")
+
+        if props.get("data"):
+            data.update(props.get("data"))
+
+        if props.get("files"):
+            files.update(props.get("files"))
+    else:
+        files["workflowSource"] = open(file, "rb")
+
+    return {"data": data, "files": files}
+
+
+def parse_workflow_zip_file(file, workflow_type):
+    """
+    Processes a workflow zip bundle
+
+    :param file: String or Path-like path to a workflow.zip file
+    :param workflow_type: String, type of workflow to expect (e.g. "wdl")
+
+    :rtype: dict of `data` and `files`
+
+    If the zip only contains a single file, that file is set as `workflowSource`
+
+    If the zip contains multiple files with a MANIFEST.json file, the MANIFEST is used to determine
+    appropriate `data` and `file` arguments. (See: parse_workflow_manifest_file())
+
+    If the zip contains multiple files without a MANIFEST.json file:
+      * a `main` workflow file with an extension matching the workflow_type is expected and will be set as `workflowSource`
+      * optionally, if `inputs*.json` files are found in the root level of the zip, they will be set as `workflowInputs(_\d)*` in the order they are found
+      * optionally, if an `options.json` file is found in the root level of the zip, it will be set as `workflowOptions`
+
+    If the zip contains multiple files, the original zip is set as `workflowDependencies`
+    """
+    data = dict()
+    files = dict()
+
+    wd = path.dirname(file)
+    with zipfile.ZipFile(file) as zip:
+        zip.extractall(wd)
+
+        contents = zip.namelist()
+        if not contents:
+            raise RuntimeError("empty workflow.zip")
+
+        if len(contents) == 1:
+            # single file workflow
+            files["workflowSource"] = open(path.join(wd, contents[0]), "rb")
+
+        else:
+            # multifile workflow
+            if "MANIFEST.json" in contents:
```

Comment:
I'm not seeing support for multi file workflows in the Batch/Nextflow/MiniWdl adapters. Does it just work out of the box or can those engines not support a MANIFEST.json?

Reply:
Right, so the difference here is that Nextflow/Miniwdl handle that within the head job container. Cromwell adapter is quite different compared to the other two we have.